### PR TITLE
Move problem summary report to ${buildDir}/reports/problems

### PR DESCRIPTION
**Pull request by donat**
Fixes https://github.com/gradle/gradle/issues/30187

### CODE REVIEW

File: platforms/core-configuration/configuration-problems-base/src/main/kotlin/org/gradle/internal/configuration/problems/CommonReport.kt [hunk 1]
```diff
@@ -229,8 +229,10 @@ class CommonReport(
                 val reportDir = outputDirectory.resolve(reportHash())
                 val reportFile = reportDir.resolve("$reportFileName.html")
                 if (!reportFile.exists()) {
-                    require(reportDir.mkdirs()) {
-                        "Could not create $reportFileName directory '$reportDir'"
+                    if (!reportDir.exists()) {
+                        require(reportDir.mkdirs()) {
+                            "Could not create $reportFileName directory '$reportDir'"
+                        }
```
Comment (donat) [inline platforms/core-configuration/configuration-problems-base/src/main/kotlin/org/gradle/internal/configuration/problems/CommonReport.kt:235]: Slightly unrelated change. This handles the case where the user manually deletes the report file but keeps the container directory. Then, the next time the report generation is invoked, the build would fail because mkdirs()` returns false.

**Comment by donat** [comment]
@bot-gradle test this

**Comment by donat** [comment]
@bot-gradle test this

**Comment by donat** [comment]
@bot-gradle test this

**Comment by donat** [comment]
@bot-gradle test this

**Review by reinsch82 — APPROVED** [review]
LGTM

**Comment by donat** [comment]
@bot-gradle merge
